larpandoracontent/LArCheating/CheatingBeamParticleIdTool.cc

```diff
@@ -0,0 +1,82 @@
+/**
+ *  @file   larpandoracontent/LArCheating/CheatingBeamParticleIdTool.cc
+ *
+ *  @brief  Implementation of the cheating beam particle id tool class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArCheating/CheatingBeamParticleIdTool.h"
+
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+CheatingBeamParticleIdTool::CheatingBeamParticleIdTool() : 
+    m_minWeightFraction(0.5f)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void CheatingBeamParticleIdTool::SelectOutputPfos(const pandora::Algorithm *const /*pAlgorithm*/, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, PfoList &selectedPfos)
+{
+    if (nuSliceHypotheses.size() != crSliceHypotheses.size())
+        throw StatusCodeException(STATUS_CODE_INVALID_PARAMETER);
+
+    IntVector beamSliceIndices;
```

Comment:
This could be a `std::set` of integers, given that we're only looking to see (later) if a given integer is in the set. However, no real problem with this being a vector as the number of slices is likely to only be of the order of a few tens.

Reply:
Removed in restructuring of code.